src/lib/core/say.c

```diff
@@ -342,10 +342,14 @@ static int
 logrotate_cb(struct coio_task *ptr)
 {
 	struct rotate_task *task = (struct rotate_task *) ptr;
-	if (log_rotate(task->log) < 0) {
+	struct log *log = task->log;
+	if (log_rotate(log) < 0)
 		diag_log();
-	}
-	ev_async_send(task->loop, &task->log->log_async);
+	tt_pthread_mutex_lock(&log->mutex);
+	log->rotating_threads--;
```

Comment:
Nit: It'd be nice to have an assertion here that `rotating_threads` never goes below 0.